langchain4j-vertex-ai/src/main/java/dev/langchain4j/model/vertexai/internal/IndexEndpointService.java

```diff
@@ -0,0 +1,152 @@
+package dev.langchain4j.model.vertexai.internal;
+
+import com.google.api.gax.core.CredentialsProvider;
+import com.google.cloud.aiplatform.v1.*;
+import lombok.Getter;
+import lombok.NonNull;
+import lombok.experimental.SuperBuilder;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.lang3.StringUtils;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.stream.Collectors;
+
+@Slf4j
+@SuperBuilder
+public class IndexEndpointService {
+
+    @NonNull
+    private final String endpoint;
+    @NonNull
+    private final String location;
+    @NonNull
+    private final String project;
+    private final String indexEndpointId;
+    private final String indexId;
+    @Getter(lazy = true)
+    private final IndexServiceSettings indexServiceSettings = initIndexServiceClientSettings();
+    private final CredentialsProvider credentialsProvider;
+    @Getter(lazy = true)
+    private final String publicEndpoint = initPublicEndpoint();
+
+    /**
+     * Deletes the embedding index.
+     *
+     * @param embeddingIndices the embedding indices
+     */
+    public void deleteIndices(List<String> embeddingIndices) {
+        if (StringUtils.isEmpty(indexId)) {
+            log.warn("Index is not specified, skipping delete.");
```

Comment:
this should fail in my opinion, you are using this in tests only

Reply:
It was but the index may not exists. Will push some adding feature later like index create. But before creating an index we need to have file in the GCP other the index cannot be created.